hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieBaseFileGroupRecordBuffer.java

```diff
@@ -253,10 +261,10 @@ protected Option<Pair<T, Map<String, Object>>> doProcessNextDataRecord(T record,
             } else {
               Option<Pair<HoodieRecord, Schema>> combinedRecordAndSchemaOpt = recordMerger.get().merge(
                   readerContext.constructHoodieRecord(Option.of(record), metadata),
-                  (Schema) metadata.get(INTERNAL_META_SCHEMA),
+                  readerContext.getSchemaHandler().decodeAvroSchema(metadata.get(INTERNAL_META_SCHEMA_ID)),
                   readerContext.constructHoodieRecord(
                       existingRecordMetadataPair.getLeft(), existingRecordMetadataPair.getRight()),
-                  (Schema) existingRecordMetadataPair.getRight().get(INTERNAL_META_SCHEMA),
+                  readerContext.getSchemaHandler().decodeAvroSchema(existingRecordMetadataPair.getRight().get(INTERNAL_META_SCHEMA_ID)),
```

Comment:
instead of having readerContext.getSchemaHandler().decodeAvroSchema(
make a method in the schema handler getSchemaFromRecordMetadata() that takes in the map instead of calling get(INTERNAL_META_SCHEMA_ID) a bunch of times

Reply:
Also not even sure if this should be in the schema handler. So maybe I can just refactor it later